src/components/picker/PickerItem.tsx

```diff
@@ -44,7 +44,7 @@ const PickerItem = (props: PickerItemProps) => {
 
   const isItemDisabled = useMemo(() => {
     return !!(disabled || (!isSelected && context.selectionLimit && context.selectionLimit === selectedCounter));
-  }, [selectedCounter]);
+  }, [selectedCounter, disabled]);
```

Comment:
Is this related to the main issue?
Why not add the other missing deps?

Reply:
yes it's related.
If the `disabled` is not in the deps of the `useMemo` then the search return the disabled item as selectable item.

Reply:
about the other missing deps, I'll add `isSelected`.
Not sure if I should add the context we can talk about it.